src/script/lua_api/l_object.cpp

```diff
@@ -542,16 +551,128 @@ int ObjectRef::l_get_bone_position(lua_State *L)
 	ServerActiveObject *sao = getobject(ref);
 	if (sao == nullptr)
 		return 0;
-
 	std::string bone = readParam<std::string>(L, 2, "");
+	BoneOverride *override = sao->getBoneOverride(bone);
+	push_v3f(L, override->position.vector);
+	v3f euler_rot;
+	override->rotation.next.toEuler(euler_rot);
+	push_v3f(L, euler_rot * core::RADTODEG);
+	return 2;
+}
 
-	v3f position = v3f(0, 0, 0);
-	v3f rotation = v3f(0, 0, 0);
-	sao->getBonePosition(bone, &position, &rotation);
+// set_bone_override(self, bone, override)
+int ObjectRef::l_set_bone_override(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == NULL)
+		return 0;
+	std::string bone;
+	if (!lua_isnil(L, 2))
+		bone = readParam<std::string>(L, 2);
+	BoneOverride *override = new BoneOverride();
+	if (lua_isnil(L, 3) || lua_isnone(L, 3)) {
+		sao->setBoneOverride(bone, override);
+		return 0;
+	}
+	lua_getfield(L, 3, "position");
+	if (!lua_isnil(L, -1)) {
+		override->position.vector = check_v3f(L, 4);
```

Comment:
Invalid index here and for absolute.

Suggestion: Replace the common `"absolute"` and `"interpolation_duration"` code with a Lambda to reduce redundancy and inconsistency.